package.json

```diff
@@ -25,27 +26,43 @@
     "@types/debug": "0.0.30",
     "@types/dotenv": "^4.0.3",
     "@types/express": "^4.0.35",
+    "@types/jest": "^23.3.9",
     "@types/mocha": "^5.2.0",
     "@types/mongoose": "^5.0.14",
     "@types/morgan": "^1.7.32",
-    "@types/node": "^10.1.2",
+    "@types/node": "^10.12.2",
     "@types/passport": "^0.4.5",
     "@types/passport-jwt": "^3.0.1",
+    "@types/supertest": "^2.0.6",
+    "@types/typescript": "^2.0.0",
+    "babel-cli": "^6.26.0",
     "chai": "^4.0.2",
     "chai-http": "^4.0.0",
     "gulp": "^4.0.0",
     "gulp-clean": "^0.4.0",
     "gulp-typescript": "^4.0.2",
+    "gulp-watch": "^5.0.1",
     "gulp-yaml": "^2.0.1",
+    "jest": "^23.6.0",
     "mocha": "^5.2.0",
     "mocha-typescript": "^1.1.4",
+    "node-typescript": "^0.1.3",
+    "superagent": "^4.0.0-beta.5",
+    "supertest": "^3.3.0",
+    "ts-jest": "^23.10.4",
+    "tslint": "^5.11.0",
     "typescript": "^2.3.4"
   },
   "scripts": {
     "start": "./node_modules/.bin/nodemon dist/index.js",
+    "lint-server": "tslint -c tslint.json 'server/**/*.{ts,tsx}'",
+    "lint-tests": "tslint -c tslint.json 'tests/**/*.{ts,tsx}'",
+    "pretest": "npm run lint-server",
     "build": "./node_modules/.bin/gulp build",
-    "test": "./node_modules/.bin/mocha -t 30000 dist/tests/**/*.spec.js",
-    "postinstall": "npm run build"
+    "test-mocha": "./node_modules/.bin/mocha -t 30000 dist/tests/**/*.spec.js",
```

Comment:
Combine this with the test command below. Also don't need to prefix this with `./node_modules/`